test/unit/requestWrapper.test.js

```diff
@@ -1,6 +1,279 @@
 'use strict';
-
+const fs = require('fs');
 const formatError = require('../../lib/requestwrapper').formatError;
+const sendRequest = require('../../lib/requestwrapper').sendRequest;
+
+jest.mock('axios');
+const axios = require('axios');
+
+describe('sendRequest', () => {
+  afterEach(() => {
+    axios.mockClear();
+  });
+
+  it('should send a request with default parameters', done => {
+    const parameters = {
+      defaultOptions: {
+        body: 'post=body',
+        formData: '',
+        qs: {},
+        method: 'POST',
+        rejectUnauthorized: true,
+        url:
+          'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+        responseType: 'jsonp',
+      },
+    };
+
+    axios.mockResolvedValue('res');
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(axios.mock.calls[0][0].data).toEqual('post=body');
+      expect(axios.mock.calls[0][0].url).toEqual(
+        'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id'
+      );
+      expect(axios.mock.calls[0][0].headers).toEqual({
+        'Accept-Encoding': 'gzip',
+        'test-header': 'test-header-value',
+      });
+      expect(axios.mock.calls[0][0].httpsAgent.options.rejectUnauthorized).toEqual(
+        parameters.defaultOptions.rejectUnauthorized
+      );
+      expect(axios.mock.calls[0][0].method).toEqual(parameters.defaultOptions.method);
+      expect(axios.mock.calls[0][0].responseType).toEqual(parameters.defaultOptions.responseType);
+      expect(res).toEqual('res');
+      expect(axios.mock.calls.length).toBe(1);
+      done();
+    });
+  });
+
+  it('should call formatError if request failed', done => {
+    const parameters = {
+      defaultOptions: {
+        body: 'post=body',
+        formData: '',
+        qs: {},
+        method: 'POST',
+        rejectUnauthorized: true,
+        url:
+          'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+        responseType: 'json',
+      },
+    };
+
+    axios.mockRejectedValue('error');
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(err).toEqual(expect.anything());
+      expect(body).toBeUndefined();
+      expect(res).toBeUndefined();
+      done();
+    });
+  });
+
+  it('should send a request with overwritten parameters', done => {
```

Comment:
I think it would be helpful to add a comment somewhere in this test stating that `options` should have the priority over `defaultOptions` for clarity